Fix missing sounds when interacting with item frames

zSALLAZAR opened the pull request:
## Introduction

### Relevant issues

#5168 

## Changes
### API changes

### Behavioural changes

## Backwards compatibility

## Follow-up

## Tests

https://user-images.githubusercontent.com/59490940/199596126-a545f8cc-c153-4112-a344-55cbea54ea20.mp4

Commit:
Fix missing sounds when interacting with item frames

Commit:
Wrong sound and missing sound

JavierLeon9966 commented on src/network/mcpe/handler/InGamePacketHandler.php:
```diff
@@ -736,7 +737,9 @@ public function handleItemFrameDropItem(ItemFrameDropItemPacket $packet) : bool{
 		$block = $this->player->getWorld()->getBlockAt($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ());
 		if($block instanceof ItemFrame && $block->getFramedItem() !== null){
-			return $this->player->attackBlock(new Vector3($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ()), $block->getFacing());
+			$this->player->getWorld()->addSound($pos = new Vector3($blockPosition->getX(), $blockPosition->getY(), $blockPosition->getZ()), new ItemFrameRemoveItemSound());
```
This should be in `ItemFrame->onAttack()`, you don't know if the item frame is empty

JavierLeon9966 replied:
Now that I see that code above it it does check, but you don't know if a plugin has canceled the interact action

zSALLAZAR replied:
You're right
I'll change it tomorrow

Commit:
Fix and add missing sound

dktapps commented on src/block/ItemFrame.php:
```diff
@@ -162,6 +171,7 @@ public function onAttack(Item $item, int $face, ?Player $player = null) : bool{
 	public function onNearbyBlockChange() : void{
 		if(!$this->getSide(Facing::opposite($this->facing))->isSolid()){
 			$this->position->getWorld()->useBreakOn($this->position);
+			$this->position->getWorld()->addSound($this->position, new ItemFrameRemoveSound());
```
`ItemFrameDestroySound` would be more consistent, though I don't see why item frames need a dedicated sound for this. We don't play specific destroy sounds or particles for nearby updates for any other block.

zSALLAZAR replied:
If you remove the block behind the itemframe in vanilla a itemframe destroy sound is played, but not in Pocketmine, so I added this sound.

dktapps commented:
Retargeting this to next-minor since it's adding new APIs.

Commit:
Rename sound

dktapps committed:
Remove destroy sounds
this should be handled by useBreakOn(), and missing destroy sounds like this are part of a larger bug.

dktapps committed:
Merge branch 'next-minor' into fix-missing-sounds-when-interacting-with-item-frames

dktapps commented:
I've removed the onNearbyBlockChange changes since those don't directly pertain to interaction, and the missing sounds there are part of a larger issue which should be solved in a more generic way. The rest of the PR is good to merge though.
